cli/src/config/misc.toml

```diff
@@ -13,7 +13,7 @@ allow-filesets = false
 always-allow-large-revsets = false
 diff-instructions = true
 paginate = "auto"
-pager = { command = ["less", "-FRX"], env = { LESSCHARSET = "utf-8" } }
+pager = { command = ["less"], env_default = { LESS="-FRX", LESSCHARSET = "utf-8" } }
```

Comment:
Isn't it better to override the `LESS` variable as before?

I don't remember exactly, but some default-installed `~/.bashrc` might set `LESS=..`. If it didn't contain `-R`, jj's default would break.

Reply:
Firstly, I think you don't need the `-` in LESS.

I'm not sure about Yuya's point. I have LESS set to `FRX --mouse --wheel-lines=2`, and I wouldn't like it overridden entirely. Perhaps we could print a warning after `less` quits if `FRX` was not in LESS and it was set? Perhaps we could prepend FRX to the variable?

Reply:
I also think that, while it might make sense to add `-R` to LESS, forcibly adding `-F` (and possibly also `-X`) might be wrong if the user explicitly did not include them. For example, let's say we launch `jj` from a file manager like Midnight Commander or ranger. The user might then be relying on `less` not quitting after one screen to see the output at all. If we forcibly add `-F`, that would break this use-case. Forcibly adding `-X` might break it too, but I'm less sure.

So, in my opinion, if we want to hack around weird default values for LESS, we could do the analogue of `LESS="R $LESS"`, but probably no more than that.

Reply:
I now think that if some default `~/.bashrc` sets LESS to some value that doesn't contain `R`, that's a bug in that default config. I don't think it happens often in practice.

So, it sounds like I'm converging to something that's very close to this PR as is. (See also https://github.com/martinvonz/jj/pull/3657/files#r1833620765).

One thing I'd now do differently is to move `pager.env_default` to `pager_env_default`, outside of `pager`, so `ui.pager` is a string again (by default, at least). If we set these environment variables whenever they are unset, it might be more consistent to set them even if the user sets `ui.pager="less"` on Windows. But the difference between that and the current behavior of the PR is not so great, it's more of a minor detail.